testing/common/data/record/record_test.go

```diff
@@ -0,0 +1,763 @@
+package record
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"kantoku/common/data/record"
+	"kantoku/impl/common/data/record/dumb"
+	mongorec "kantoku/impl/common/data/record/mongo"
+	"log"
+	"math/rand"
+	"sort"
+	"testing"
+)
+
+func newMongoRecords(ctx context.Context) *mongorec.Storage {
+	// Set connection configurations
+	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017")
+
+	// Connect to the MongoDB server
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+	go func() {
+		<-ctx.Done()
+
+		if err = client.Disconnect(ctx); err != nil {
+			log.Println("failed to disconnect from mongodb:", err)
+		}
+	}()
+
+	// Access the "main" database
+	database := client.Database("testing")
+
+	// Access the "testing" collection
+	collection := database.Collection("records")
+
+	return mongorec.New(collection)
+}
+
+var Keys = []string{
+	"updated_at",
+	"id",
+	"name",
+	"surname",
+	"group",
+}
+
+var Groups = []string{
+	"a",
+	"b",
+	"c",
+	"d",
+	"e",
+}
+
+const MinimumTestCases = 5
+const MaximumTestCases = 15
+const TotalRecords = 100
+
+func TestRecord(t *testing.T) {
```

Comment:
soo, i'm afraid we need simple hand-coded tests to run on both of them, to check that they actually do something

Reply:
Added some simple cases

Reply:
I refactored a bit. I check all impls with hand tests. Also moved tests from function which iterates over impls.